starter-code/src/clue.js

```diff
@@ -1,94 +1,161 @@
-occupation:   Socialité
-
-mrMustard
-first_name:   Jack
-last_name:    Mustard
-color:        yellow
-description:  He is a former football player who tries to get by on his former glory
-age:          62
-image:        https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg
-occupation:   Retired Football player
-
-// Weapons
-
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+var rope = { name: "rope", weight: 10 };
+var knife = { name: "knife", weight: 8 };
+var candlestick = { name: "andlestick", weight: 2 };
+var dumbbell = { name: "dumbbell", weight: 30 };
+var poison = { name: "poison", weight: 2 };
+var axe = { name: "axe", weight: 15 };
+var bat = { name: "bat", weight: 13 };
+var trophy = { name: "trophy", weight: 25 };
+var pistol = { name: "pistol", weight: 20 };
 
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+var DinningRoom = { name: "Dinning Room" };
+var Conservatory = { name: "Conservatory" };
+var Kitchen = { name: "Kitchen" };
+var Study = { name: "Study" };
+var Library = { name: "Library" };
+var BilliardRoom = { name: "Billiard Room" };
+var Lounge = { name: "Lounge" };
+var Ballroom = { name: "Ballroom" };
+var Hall = { name: "Hall" };
+var Spa = { name: "Spa" };
+var LivingRoom = { name: "Living Room" };
+var Observatory = { name: "Observatory" };
+var Theater = { name: "Theater" };
+var GuestHouse = { name: "Guest House" };
+var Patio = { name: "Patio" };
 
 // Characters Collection
 var charactersArray = [];
+charactersArray.push(
+  mrGreen,
+  drOrchid,
+  profPlum,
+  missScarlet,
+  mrsPeacock,
+  mrMustard
+);
 
 // Rooms' Collection
 var roomsArray = [];
+roomsArray.push(
+  DinningRoom,
+  Conservatory,
+  Kitchen,
+  Study,
+  Library,
+  BilliardRoom,
+  Lounge,
+  Ballroom,
+  Hall,
+  Spa,
+  LivingRoom,
+  Observatory,
+  Theater,
+  GuestHouse,
+  Patio
+);
 
 // Weapons Collection
 var weaponsArray = [];
+weaponsArray.push(
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+);
+console.log(weaponsArray);
+
+function randomSelector(array) {
+  var index = Math.floor(Math.random() * array.length);
+  return array[index];
+}
+console.log(randomSelector(weaponsArray));
+
+function pickMistery() {
+  var misteryEnvelope = [];
+  misteryEnvelope.push(
+    randomSelector(charactersArray),
+    randomSelector(weaponsArray),
+    randomSelector(roomsArray)
+  );
+  return misteryEnvelope;
+}
+function revealMistery(misteryEnvelope) {
```

Comment:
what you've done here is almost correct, but you just need to make a few changes. 

```
function revealMistery(misteryEnvelope) {
  return (mystery =
    misteryEnvelope[0].first_name +
    misteryEnvelope[0].last_name +
    " killed Mr.Boddy using the " +
    misteryEnvelope[1].name +
    " in the " +
    misteryEnvelope[2].name);
}
console.log(revealMistery(misteryEnvelope));

```


you passed the misteryEnvelope because that already contains what you need (the murderer, weapon and room) so you  don't need to pickMistery again. Here you just needed to return the final result using what you have already worked out 

another small thing- for the weapon you used the same index as for the room